app/irc.py

```diff
@@ -314,34 +364,43 @@ async def send_message(self, fro: Player, to: str, message: str) -> int:
 
 class IRCServer:
     def __init__(self, port: int, loop: asyncio.AbstractEventLoop) -> None:
-        self.loop: asyncio.AbstractEventLoop = loop
-        self.host: str = "0.0.0.0"
-        self.port: int = port
-        self.socket_server: asyncio.Server = None
+        self.loop = loop
+        self.host = "0.0.0.0"
+        self.port = port
+        self.socket_server: Optional[asyncio.Server] = None
         self.clients: set[IRCClient] = set()
 
     @property
-    def authorized_clients(self) -> list[IRCClient]:
+    def authorized_clients(self) -> set[IRCClient]:
         return {x for x in self.clients if x.player is not None}
 
     def bancho_join(self, player: Player, channel: Channel) -> None:
         for client in self.authorized_clients:
+            assert client.player is not None
+
             if channel in client.player.channels:
                 client.add_queue(f":{player.name} JOIN {channel.name}")
 
     def bancho_part(self, player: Player, channel: Channel) -> None:
         for client in self.authorized_clients:
+            assert client.player is not None
+
             if channel in client.player.channels:
                 client.add_queue(f":{player.name} PART {channel.name}")
 
     def bancho_message(self, fro: str, to: str, message: str) -> None:
         if to.startswith("#"):
             for client in self.authorized_clients:
-                if client.player.name != fro:
-                    to in [x.name for x in client.player.channels]
+                assert client.player is not None
+
+                if client.player.name != fro and to in [
+                    x.name for x in client.player.channels
+                ]:
```

Comment:
@7mochi could you confirm if this is correct?

there was a bug here with `to in [x.name for x in client.player.channels]` not being used in the `if` statement. I assumed this was the correct fix.

Reply:
yep 👍